<a href="https://colab.research.google.com/github/Pam2020/DATA602-FinalProject-ChicagoEnergy/blob/main/assignment_04/assignment_04_SravyaPamula.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Assignment 4

Name: Sravya Pamula

### Question 6:

After downloading the data, we just pick the CSV file that contains data of 2017 and 2018 data. We use pandas library to work with this data.

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
df = pd.read_csv('/content/MERGED2017_18_PP.csv')
df.head(5)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (1729) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


UNITID    OPEID  OPEID6                               INSTNM        CITY  \
0  100654   100200    1002             Alabama A & M University      Normal   
1  100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2  100690  2503400   25034                   Amridge University  Montgomery   
3  100706   105500    1055  University of Alabama in Huntsville  Huntsville   
4  100724   100500    1005             Alabama State University  Montgomery   

  STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0     AL       35762           NaN      NaN     NaN  ...   
1     AL  35294-0110           NaN      NaN     NaN  ...   
2     AL  36117-3553           NaN      NaN     NaN  ...   
3     AL       35899           NaN      NaN     NaN  ...   
4     AL  36104-0271           NaN      NaN     NaN  ...   

   OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN   G12MN  \
0                           NaN       0.7106        0.7418   5207.0  1185.0   
1                           NaN       0.4009        0.5488  13308.0  9888.0   
2                           NaN       0.5000        0.5000    431.0   437.0   
3                           NaN       0.2012        0.4231   7519.0  2378.0   
4                           NaN       0.7594        0.7402   5229.0   687.0   

   SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0   1410.0                 NaN                      NaN   
1   1948.0                 NaN                      NaN   
2      4.0                 NaN                      NaN   
3   1203.0                 NaN                      NaN   
4   1143.0                 NaN                      NaN   

   FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                       NaN             NaN  
1                       NaN             NaN  
2                       NaN             NaN  
3                       NaN             NaN  
4                       NaN             NaN  

[5 rows x 1986 columns]

We have 7112 rows in our dataset.  When we check the number of unique institute names in the INSTNM column we get a total of 6974 colleges. There are 1986 columns/variables in the given dataset.

In [ ]:
df.shape
# shape gives the rows and columns in a dataframe in the format (rows, columns).

(7112, 1986)

In [ ]:
df['INSTNM'].nunique()
# nunique() will give the number of unique entries in the INSTNM column

6974

Here we look at the following:
1. First 5 colleges - using head()
2. Last 5 colleges - using tail()
3. Random 5 colleges - using sample()

In [53]:
df.head(5)

UNITID    OPEID  OPEID6                               INSTNM        CITY  \
0  100654   100200    1002             Alabama A & M University      Normal   
1  100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2  100690  2503400   25034                   Amridge University  Montgomery   
3  100706   105500    1055  University of Alabama in Huntsville  Huntsville   
4  100724   100500    1005             Alabama State University  Montgomery   

  STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0     AL       35762           NaN      NaN     NaN  ...   
1     AL  35294-0110           NaN      NaN     NaN  ...   
2     AL  36117-3553           NaN      NaN     NaN  ...   
3     AL       35899           NaN      NaN     NaN  ...   
4     AL  36104-0271           NaN      NaN     NaN  ...   

   OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN   G12MN  \
0                           NaN       0.7106        0.7418   5207.0  1185.0   
1                           NaN       0.4009        0.5488  13308.0  9888.0   
2                           NaN       0.5000        0.5000    431.0   437.0   
3                           NaN       0.2012        0.4231   7519.0  2378.0   
4                           NaN       0.7594        0.7402   5229.0   687.0   

   SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0   1410.0                 NaN                      NaN   
1   1948.0                 NaN                      NaN   
2      4.0                 NaN                      NaN   
3   1203.0                 NaN                      NaN   
4   1143.0                 NaN                      NaN   

   FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                       NaN             NaN  
1                       NaN             NaN  
2                       NaN             NaN  
3                       NaN             NaN  
4                       NaN             NaN  

[5 rows x 1986 columns]

In [54]:
df.tail(5)

UNITID    OPEID  OPEID6  \
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR        ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
7107     AZ  853656900           NaN      NaN     NaN  ...   
7108     CA      95113           NaN      NaN     NaN  ...   
7109     CA      93301           NaN      NaN     NaN  ...   
7110     MA  021480000           NaN      NaN     NaN  ...   
7111     FL      33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
7107                           NaN          NaN           NaN     NaN    NaN   
7108                           NaN          NaN           NaN     NaN    NaN   
7109                           NaN          NaN           NaN     NaN    NaN   
7110                           NaN          NaN           NaN     NaN    NaN   
7111                           NaN          NaN           NaN     NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
7107      NaN                 NaN                      NaN   
7108      NaN                 NaN                      NaN   
7109      NaN                 NaN                      NaN   
7110      NaN                 NaN                      NaN   
7111      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
7107                       NaN             NaN  
7108                       NaN             NaN  
7109                       NaN             NaN  
7110                       NaN             NaN  
7111                       NaN             NaN  

[5 rows x 1986 columns]

In [55]:
df.sample(n = 5)

UNITID    OPEID  OPEID6  \
6205    482459  1072712   10727   
1645    169220  2198900   21989   
7081  45920401  4169701   41697   
4714    431770  1095200   10952   
5852    461625  4176000   41760   

                                                 INSTNM              CITY  \
6205                           DeVry University-Florida           Miramar   
1645                      Michigan School of Psychology  Farmington Hills   
7081                                     Unitek College           Fremont   
4714  Putnam Westchester BOCES-Practical Nursing Pro...  Yorktown Heights   
5852                       American Technical Institute           Bayamon   

     STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
6205     FL       33027           NaN      NaN     NaN  ...   
1645     MI  48334-4512           NaN      NaN     NaN  ...   
7081     CA   945383185           NaN      NaN     NaN  ...   
4714     NY  10598-4399           NaN      NaN     NaN  ...   
5852     PR  00957-0000           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
6205                           NaN       0.7692        0.6923  1377.0  673.0   
1645                           NaN          NaN           NaN     NaN  185.0   
7081                           NaN          NaN           NaN     NaN    NaN   
4714                           NaN       0.4000        0.8500    43.0    NaN   
5852                           NaN       1.0000        0.0000    77.0    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
6205     13.0                 NaN                      NaN   
1645      NaN                 NaN                      NaN   
7081      NaN                 NaN                      NaN   
4714     20.0                 NaN                      NaN   
5852     36.0                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
6205                       NaN             NaN  
1645                       NaN             NaN  
7081                       NaN             NaN  
4714                       NaN             NaN  
5852                       NaN             NaN  

[5 rows x 1986 columns]

### Question 7:

Here we will create a new dataframe with just 9 of the following columns: 
1. UNITID
2. INSTNM
3. CITY
4. STABBR
5. ZIP
6. ADM_RATE
7. UGDS
8. TUITIONFEE_IN
9. MN_EARN_WNE_P10


In [ ]:
df_filtered = df[['UNITID', 'INSTNM', 'CITY', 'STABBR', 'ZIP', 'ADM_RATE', 'UGDS', 'TUITIONFEE_IN', 'MN_EARN_WNE_P10']]
df_filtered.head()

UNITID                               INSTNM        CITY STABBR         ZIP  \
0  100654             Alabama A & M University      Normal     AL       35762   
1  100663  University of Alabama at Birmingham  Birmingham     AL  35294-0110   
2  100690                   Amridge University  Montgomery     AL  36117-3553   
3  100706  University of Alabama in Huntsville  Huntsville     AL       35899   
4  100724             Alabama State University  Montgomery     AL  36104-0271   

   ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0    0.9027   4824.0         9857.0              NaN  
1    0.9181  12866.0         8328.0              NaN  
2       NaN    318.0         6900.0              NaN  
3    0.8123   6917.0        10280.0              NaN  
4    0.9787   4189.0        11068.0              NaN

### Question 8 and 9: 

We look at the data dictionary  and find what these 9 chosen variables mean.

1. UNITID - A unique identification number assigned to postsecondary institutions through the IPEDS survey.

2. INSTNM - The institution name as reported by IPEDS. 

3. CITY - The city where the institute is located as reported by IPEDS. 

4. STABBR - The state where the institute is located as reported by IPEDS. 

5. ZIP -  Institute zipcode. 

6. ADM_RATE -  This is the Fall admissions rate at that campus defined as the number of undergraduates admitted by the number of undergraduate admissions. 

7. UGDS - The number of degree/certificate seeking undergraduate students enrolled in the institute. 

8. TUITIONFEE_IN - Tuition and required fees for in-state students in the given institute. 

9. MN_EARN_WNE_P10 - Mean earnings of graduates working 10 years after the entry. This column seems to have only NaN values. 



In [ ]:
df_filtered['MN_EARN_WNE_P10'].unique()

array([nan])

### Question 10:

Exploring the smaller data frame:

1)  Look at the first 10 colleges


In [ ]:
df_filtered.head(10)

UNITID                               INSTNM            CITY STABBR  \
0  100654             Alabama A & M University          Normal     AL   
1  100663  University of Alabama at Birmingham      Birmingham     AL   
2  100690                   Amridge University      Montgomery     AL   
3  100706  University of Alabama in Huntsville      Huntsville     AL   
4  100724             Alabama State University      Montgomery     AL   
5  100751            The University of Alabama      Tuscaloosa     AL   
6  100760    Central Alabama Community College  Alexander City     AL   
7  100812              Athens State University          Athens     AL   
8  100830      Auburn University at Montgomery      Montgomery     AL   
9  100858                    Auburn University          Auburn     AL   

          ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0       35762    0.9027   4824.0         9857.0              NaN  
1  35294-0110    0.9181  12866.0         8328.0              NaN  
2  36117-3553       NaN    318.0         6900.0              NaN  
3       35899    0.8123   6917.0        10280.0              NaN  
4  36104-0271    0.9787   4189.0        11068.0              NaN  
5  35487-0100    0.5330  32387.0        10780.0              NaN  
6       35010       NaN   1404.0         4440.0              NaN  
7       35611       NaN   2801.0            NaN              NaN  
8  36117-3596    0.8240   4211.0         8020.0              NaN  
9       36849    0.8393  23391.0        10968.0              NaN

2)  Look at the last 10 colleges

In [ ]:
df_filtered.tail(10)

UNITID                                             INSTNM  \
7102  48154401  National Personal Training Institute of Cleveland   
7103  48305401             Barber School of Pittsburgh - Ambridge   
7104  48312401                 Arizona State University at Tucson   
7105  48312402  Arizona State University - ASU Colleges at Lak...   
7106  48312403        Arizona State University at The Gila Valley   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                  CITY STABBR        ZIP  ADM_RATE  UGDS  TUITIONFEE_IN  \
7102  Highland Heights     OH      44143       NaN   NaN            NaN   
7103          Ambridge     PA  150030000       NaN   NaN            NaN   
7104            Tucson     AZ  857454284       NaN   NaN            NaN   
7105  Lake Havasu City     AZ  864036877       NaN   NaN         6904.0   
7106          Thatcher     AZ  855525545       NaN   NaN            NaN   
7107              Yuma     AZ  853656900       NaN   NaN            NaN   
7108          San Jose     CA      95113       NaN   NaN            NaN   
7109       Bakersfield     CA      93301       NaN   NaN            NaN   
7110            Malden     MA  021480000       NaN   NaN            NaN   
7111   Fort Lauderdale     FL      33309       NaN   NaN        22575.0   

      MN_EARN_WNE_P10  
7102              NaN  
7103              NaN  
7104              NaN  
7105              NaN  
7106              NaN  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN

3)  Look at a random sample of 10 colleges.

In [ ]:
df_filtered.sample(10)

UNITID                                      INSTNM          CITY  \
1543    165635    National Aviation Academy of New England       Concord   
3924    241739      Universidad Ana G. Mendez-Cupey Campus      San Juan   
6801  16884701                     Baker College of Owosso        Owosso   
5422    453163           Strayer University-North Carolina   Morrisville   
5037    444200      Harris School of Business-Dover Campus         Dover   
1140    152390                        Saint Mary's College    Notre Dame   
1009    147031        Meadville Lombard Theological School       Chicago   
6310    484358  Ross Medical Education Center-Johnson City  Johnson City   
586     128744                    University of Bridgeport    Bridgeport   
5668    458405     Platt College-Miller-Motte-Fayetteville  Fayetteville   

     STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1543     MA       01742       NaN    189.0            NaN              NaN  
3924     PR  00926-2602       NaN  10617.0         5820.0              NaN  
6801     MI   488679600       NaN      NaN         9360.0              NaN  
5422     NC       27560       NaN   3980.0        13857.0              NaN  
5037     DE       19901       NaN    226.0            NaN              NaN  
1140     IN       46556    0.7825   1489.0        40800.0              NaN  
1009     IL       60605       NaN      NaN            NaN              NaN  
6310     TN  37601-1526       NaN    134.0            NaN              NaN  
586      CT  06604-5620    0.5389   3129.0        32250.0              NaN  
5668     NC       28311       NaN    658.0        10980.0              NaN

4) Find and display UMBC: We first find the colleges in Baltimore and look for UMBC in the list. Once we find the way teh name is defined w

In [ ]:
df_filtered[df_filtered['CITY'] == 'Baltimore'] 

UNITID                                   INSTNM       CITY STABBR  \
1451    161855          Baltimore Studio of Hair Design  Baltimore     MD   
1452    161864         Baltimore City Community College  Baltimore     MD   
1453    161873                  University of Baltimore  Baltimore     MD   
1460    162283                  Coppin State University  Baltimore     MD   
1462    162371             North American Trade Schools  Baltimore     MD   
1466    162654                          Goucher College  Baltimore     MD   
1473    162928                 Johns Hopkins University  Baltimore     MD   
1475    163046               Loyola University Maryland  Baltimore     MD   
1477    163259         University of Maryland Baltimore  Baltimore     MD   
1478    163268  University of Maryland-Baltimore County  Baltimore     MD   
1480    163295        Maryland Institute College of Art  Baltimore     MD   
1485    163453                  Morgan State University  Baltimore     MD   
1487    163532            Ner Israel Rabbinical College  Baltimore     MD   
1489    163578        Notre Dame of Maryland University  Baltimore     MD   
1491    163736             Brightwood College-Baltimore  Baltimore     MD   
1493    163815         Maryland Beauty Academy of Essex  Baltimore     MD   
3004    212452               Faith Theological Seminary  Baltimore     MD   
4225    373784               All-State Career-Baltimore  Baltimore     MD   
4762    434672    Community College of Baltimore County  Baltimore     MD   
5314    450076               Fortis Institute-Baltimore  Baltimore     MD   
5834    461379      Holistic Massage Training Institute  Baltimore     MD   
6016    476601  Bais HaMedrash and Mesivta of Baltimore  Baltimore     MD   
7006  43018403    Strayer University-White Marsh Campus  Baltimore     MD   

             ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1451  21201-3444       NaN     46.0            NaN              NaN  
1452  21215-7893       NaN   3797.0         3074.0              NaN  
1453  21201-5720    0.8099   2856.0         8824.0              NaN  
1460  21216-3698    0.3651   2459.0         7474.0              NaN  
1462       21244       NaN    448.0            NaN              NaN  
1466  21204-2794    0.7884   1526.0        43440.0              NaN  
1473  21218-2688    0.1254   5663.0        52170.0              NaN  
1475  21210-2699    0.7528   3903.0        47560.0              NaN  
1477  21201-1627       NaN    918.0            NaN              NaN  
1478       21250    0.6021  11130.0        11518.0              NaN  
1480  21217-4134    0.6161   1743.0        46990.0              NaN  
1485  21251-0001    0.6372   6426.0         7766.0              NaN  
1487       21208    0.6418    265.0        11900.0              NaN  
1489  21210-2476    0.7140    808.0        36070.0              NaN  
1491  21227-1063       NaN    452.0            NaN              NaN  
1493       21224       NaN     80.0            NaN              NaN  
3004  21212-2624    1.0000     66.0         5870.0              NaN  
4225       21224       NaN    561.0            NaN              NaN  
4762       21237       NaN  17614.0         4022.0              NaN  
5314       21244       NaN     87.0            NaN              NaN  
5834       21218       NaN     59.0            NaN              NaN  
6016  21209-1613    1.0000     76.0        12700.0              NaN  
7006       21236       NaN      NaN        13857.0              NaN

In [ ]:
df_filtered[df_filtered['INSTNM'] == 'University of Maryland-Baltimore County']

UNITID                                   INSTNM       CITY STABBR  \
1478  163268  University of Maryland-Baltimore County  Baltimore     MD   

        ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1478  21250    0.6021  11130.0        11518.0              NaN

5) Find Univ. of Maryland College Park: 

We find the colleges in the city college park and then find the university of maryland college park. Surpisingly, that is the only college in the city college park. 

In [ ]:
df_filtered[df_filtered['CITY'] == 'College Park'] 

UNITID                               INSTNM          CITY STABBR    ZIP  \
1479  163286  University of Maryland-College Park  College Park     MD  20742   

      ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1479     0.445  29273.0        10399.0              NaN

6) Find and display Loyola College Maryland. 

We spotted Loyola university Maryland in the list of colleges in Baltimore above. We just find that row now. 

In [ ]:
df_filtered[df_filtered['INSTNM'] == 'Loyola University Maryland'] 

UNITID                      INSTNM       CITY STABBR         ZIP  \
1475  163046  Loyola University Maryland  Baltimore     MD  21210-2699   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1475    0.7528  3903.0        47560.0              NaN

7)  Find and display Johns Hopkins

We spotted Johns Hopkins University in the list of colleges in Baltimore above. We just find that row now.

In [ ]:
df_filtered[df_filtered['INSTNM'] == 'Johns Hopkins University'] 

UNITID                    INSTNM       CITY STABBR         ZIP  \
1473  162928  Johns Hopkins University  Baltimore     MD  21218-2688   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473    0.1254  5663.0        52170.0              NaN

8)  Calculate the min, max, and average in-state tution of the above four institutions.

I use the describe method to find the descriptive statistics of the instate tuition fee column. The minimum instate fee is 10399 dollars and the maximum instate fee is 52170 dollars. The mean instate tuition fee is 30411.75 dollars.

In [ ]:
colleges = ['Johns Hopkins University','Loyola University Maryland', 'University of Maryland-College Park', 'University of Maryland-Baltimore County']
df_new = df_filtered[df_filtered['INSTNM'].isin(colleges)]

In [ ]:
df_new.head()

UNITID                                   INSTNM          CITY STABBR  \
1473  162928                 Johns Hopkins University     Baltimore     MD   
1475  163046               Loyola University Maryland     Baltimore     MD   
1478  163268  University of Maryland-Baltimore County     Baltimore     MD   
1479  163286      University of Maryland-College Park  College Park     MD   

             ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473  21218-2688    0.1254   5663.0        52170.0              NaN  
1475  21210-2699    0.7528   3903.0        47560.0              NaN  
1478       21250    0.6021  11130.0        11518.0              NaN  
1479       20742    0.4450  29273.0        10399.0              NaN

In [ ]:
df_new['TUITIONFEE_IN'].describe()

count        4.000000
mean     30411.750000
std      22546.011271
min      10399.000000
25%      11238.250000
50%      29539.000000
75%      48712.500000
max      52170.000000
Name: TUITIONFEE_IN, dtype: float64

### Question 11:

Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.

In [ ]:
states = ['MD', 'VA', 'DC', 'DE', 'PA']
df_5states = df_filtered[df_filtered['STABBR'].isin(states)]

In [ ]:
df_5states['STABBR'].unique()

array(['DE', 'DC', 'MD', 'PA', 'VA'], dtype=object)

In [ ]:
df_5states.to_csv('Five_States_Colleges.csv')